[37463] Pass in work package project to IUM

oliverguenther opened the pull request:
When showing a work package from a subproject, the invite user modal should be opened for that work package's project, not the current project. Otherwise the user will not be assignable afterwards.

https://community.openproject.org/work_packages/37463

oliverguenther committed:
[37463] Pass in work package project to IUM

https://community.openproject.org/work_packages/37463

oliverguenther commented on frontend/src/app/modules/autocompleter/create-autocompleter/create-autocompleter.component.ts:
```diff
@@ -104,8 +106,11 @@ export class CreateAutocompleterComponent extends UntilDestroyedMixin implements
-        .subscribe((user: HalResource) => {
+        .pipe(
+          this.untilDestroyed(),
+          filter(user => !!user)
```
This is an unrelated fix I noticed. When cancelling the modal, the assignee field was still saved and a notification was shown.

b12f reviewed (commented):
Isn't this also used in `multi-select-edit-field` ? I was looking at this earlier today and thought I saw three distinct usages of the button

oliverguenther committed:
Add projectId from select input formly field

oliverguenther commented:
> Isn't this also used in `multi-select-edit-field` ? I was looking at this earlier today and thought I saw three distinct usages of the button

You're right. I added it to the multi select field. For the new dynamic form, we'd need a way to access the project reference of the resource we're passing in. Right now, as you can only edit in the current project, it does not need fixing there. I added a tentative check for a resource's "project" link (if any) so that would work for any resource with an explicit dependency to a project.

b12f commented on frontend/src/app/modules/invite-user-modal/button/invite-user-button.component.ts:
```diff
@@ -1,15 +1,20 @@
   styleUrls: ['./invite-user-button.component.sass']
 })
 export class InviteUserButtonComponent implements OnInit {
+  @Input() projectId:string|null;
```
Technically this can also be `undefined` since you're using optional chaining

b12f reviewed (approved):
lgtm
